devito/finite_differences/finite_difference.py

```diff
@@ -317,7 +317,10 @@ def generate_fd_shortcuts(function):
     """
     dimensions = function.indices
     space_fd_order = function.space_order
-    time_fd_order = function.time_order if function.is_TimeFunction else 0
+    try:
```

Comment:
just a reminder: we really need a mixin "Time<Something>" class that does/provides the following:
* set `is_Time` flag to True
* property for time_order
* property for time_dim
* class attribute `_time_position`
* ...

This should be inherited by:
* TimeFunction
*SparseTimeFunction